inbm/dispatcher-agent/dispatcher/source/source_command.py

```diff
@@ -41,11 +45,17 @@ def do_source_command(parsed_head: XmlHandler, os_type: OsType) -> Result:
 
     try:
         app_action = parsed_head.get_children("applicationSource")
+	url = parsed_head.get_children("applicationSource/add/gpg")["uri"]
```

Comment:
@tsirlapu  Was just looking over this PR and noticed this code. The actual execution of the command (including any checks) should not go in do_source_command. do_source_command's responsibility is to detect whether we are running an appliaction or os source command and pass it on to the _handle_app_source_command or _handle_os_source_command methods. This code should probably go in the Ubuntu/Debian specific source application add method, or should go in the abstract base class as a concrete method and then be called from each OS that needs it (currently just Ubuntu).